### DISCUSSION TITLE
How many epochs for RVC & Tortoise TTS? + Need out of memory tips

### DISCUSSION THREAD

**Discussion by HopeAndFaithStudios** [JarodMica/ai-voice-cloning#43] [General]
Hi, I'm new to AI training, I was wondering how many epochs I need and what batch size I should set on both Tortoise TTS and RVC (since I have both installed on my secondary NVIDIA laptop; including your repo), depending on the length of the dataset. My shortest model is 1 minute and 11 seconds, my longest one is 43 minutes 57 seconds. I just don't want to risk overtraining them and I really want them to sound accurate.

As for Tortoise, I was going to shorten the models to five or three -10 second segments, like it says on the GitHub page.

I also don't want to risk running into OOM (CUDA out of memory) since my secondary laptop has only 8 gigs of RAM (and possibly 4 gigs of VRAM). I really want to use my secondary laptop for this, since my primary one has AMD and is not supportive with any NVIDIA applications, and the desktop I have runs smoothly with 12 gigs of RAM, but I'd rather be portable.

I've been searching Google & Reddit **a lot**, only to find several different answers each time, so that didn't help much. Any advice would be _very_ helpful, thanks. :)

**Comment by JarodMica** [comment]
This repository actually has a validation configuration button, so that should generally set it to values that'll work for you computer.  With 4 GB of Vram, you can probably get away with 4 for your batch size and 1 for grad accum, play with these values depending on how much actual vram is being used.

Datasets I recommend at least being 10 minutes long.  Any shorter and you may not be able to get a decent model with it being generally, the more the better.  

As well, usually as long as both lines are trending downwards on the loss graph, you should be fine.  However, most of the time, you won't know until you listen to the output of the model as to whether it's good or not unfortunately.

**Reply by HopeAndFaithStudios** [reply]
Thanks for the reply! I didn't realize you were out of town till the day _after_ you posted the update on YouTube, so my mistake on that part lol. Unfortunately I cannot add more data to the datasets under 10 minutes because those sound clips are all I have. I also watched the RVC Tensorboard video you put out and that's helped me understand it more. It says on the RVC GitHub that people _have_ managed to train successfully about 1 to 2 minutes of data, so hopefully I'll be one of those people. And as for Tortoise, should I follow the steps from that GitHub as well? It says to put in five or three clips around 10 seconds. Again, thanks for the reply! :)

**Reply by JarodMica** [reply]
All good :)! Tortoise is a little different than RVC in that it tends to like more data rather than less, but you can get good stuff with 10 minutes-ish.  I use 2 reference clips, but you can use 3-5.  I usually stick with 2 because I can choose 2 really good samples on how the speaker should sound as these reference clips do have quite a large effect on inference quality.

**Reply by HopeAndFaithStudios** [reply]
Sorry for the late reply, been busy as usual, but does this mean I follow the same steps as RVC with Tortoise? Forgive me if I'm wrong, I'm just a little confused and new to this whole AI thing, but I really want to learn more about it and use it for my private projects. Just asking as a beginner, that's all.
